src/scb/user.clj

```diff
@@ -186,19 +203,75 @@
+
+(defn foo
   []
-  (let [dec (java.util.Base64/getUrlDecoder)
-        decode (fn [path]
-                 (let [filename (fs/base-name path)
-                       [^String filename _] (fs/split-ext (first (fs/split-ext filename)))]
-                   ;;(warn "decoding" filename)
-                   (String. (.decode dec filename))))
-
-        path-list (->> (core/paths :state-http-cache-path)
-                       fs/list-dir
-                       (map str)
-                       (mapv decode))]
-
-    (clojure.pprint/pprint path-list)))
+  (->> (wowi/parse-api-file-list {:url wowi/api-file-list
+                                  :response (http/download wowi/api-file-list)})
+       :parsed
+       (filterv (comp (partial utils/less-than-n-days-old? 14) :updated-date))))
+
+(defn daily-addon-update
+  "deletes the listing pages and API filelist cache,
+  refreshes addon data, downloading missing cache files as necessary,
+  finds addons updated in the last day using fresh data,
+  deletes their cache, refreshes data again.
+
+  remember: we can't trust the API filelist or listing pages to be complete, we have to consult both.
+  todo: quantify this discrepency like I did in strongbox
+  "
+  []
+  (run! delete-cache-path (core/cache-paths-matching #"page=|cat|filelist"))
+  (refresh-data)
+  (wait-for-empty-queues)
+
+  (let [age 2 ;; days. 2 days will ensure nothing is missed
+
+        updated-recently-from-listings
+        (->> (core/state-paths-matching "wowinterface/*/listing--*")
+             (group-by (comp str fs/base-name fs/parent)) ;; {"1234" [/path/to/state/1234/listing--combat-mods, ...], ...}
+             vals
+             (map first) ;; there will be 0 or many listing--* files, we want the first
+             (remove nil?) ;; if there are zero, first will give us nils
+             (map core/read-addon-data)
+             (filterv (comp (partial utils/less-than-n-days-old? age) :updated-date)))
+
+        updated-recently-from-filedetails
+        (->> (wowi/parse-api-file-list {:url wowi/api-file-list
+                                        :response (http/download wowi/api-file-list {})})
+             :parsed
+             (map :updated-date)
+             (remove nil?)
+             (filterv (partial utils/less-than-n-days-old? age)))
+
+        ;; #{ [:wowinterface 1234], [:wowinterface 4321], ... }
+        addon-id (juxt :source :source-id)
+        updated-recently (set (into (map addon-id updated-recently-from-listings)
+                                    (map addon-id updated-recently-from-filedetails)))
+
+        ;; todo: pmap this
+        _ (run! (partial apply delete-addon-cache) updated-recently)]
+
+    (refresh-data)
+    (wait-for-empty-queues)
+    (write-catalogue)
+
+    nil))
+
+(defn weekly-addon-update
+  "...? purge entire cache and refresh data? perhaps only target addons updated in the last fortnight/month.
+  this would be a big overlap but ensure changing data is "
```

Comment:
complete thought, add note: this will update download counts for those addons not otherwise updated during time period